Added tests for mouse clicks in Ex cmdline

dpelle opened the pull request:
This PR adds tests for mouse click in Ex command line:
* left or right click in command line sets position of cursor
* middle click in command line pastes

These were not yet covered by tests according to code codecov:

https://codecov.io/gh/vim/vim/src/559b9c68fe550f3af63d42e0838622aab1ceb1b3/src/ex_getln.c#L1914
https://codecov.io/gh/vim/vim/src/559b9c68fe550f3af63d42e0838622aab1ceb1b3/src/ex_getln.c#L1882

To be able to test the command line, I had to refactor a bit the mouse
helper functions in `test_termcodes.vim` to be able to use functions like
`MouseLeftClickCode()` (which only return the terminal code string) as
opposed to `MouseLeftClick()` which calls `feedkeys(...)`. with the
terminal codes string.

dpelle committed:
Added tests for mouse clicks in Ex cmdline

- left or right sets position of cursor in command line
- middle click pastes @* register in command line

dpelle committed:
Comment update only

brammool commented:
Dominique wrote:

> This PR adds tests for mouse click in Ex command line:
> * left or right click in command line sets position of cursor
> * middle click in command line pastes
> 
> These were not yet covered by tests according to code codecov:
> 
> https://codecov.io/gh/vim/vim/src/559b9c68fe550f3af63d42e0838622aab1ceb1b3/src/ex_getln.c#L1914
> https://codecov.io/gh/vim/vim/src/559b9c68fe550f3af63d42e0838622aab1ceb1b3/src/ex_getln.c#L1882
> 
> To be able to test the command line, I had to refactor a bit the mouse
> helper functions in `test_termcodes.vim` to be able to use functions like
> `MouseLeftClickCode()` (which only return the terminal code string) as
> opposed to `MouseLeftClick()` which calls `feedkeys(...)`. with the
> terminal codes string.

Thanks.  On my local system this failed with some weird characters
added.  I fixed that by using "Lx!" for the second feedkeys() argument.
But the "\<Left>" doesn't work, so I removed that.

-- 
Back up my hard drive?  I can't find the reverse switch!

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

dpelle commented:
> Thanks.  On my local system this failed with some weird characters
> added.  I fixed that by using "Lx!" for the second feedkeys() argument.
> But the "\<Left>" doesn't work, so I removed that.

OK. I had put `\<Left>` to check that paste happens at the position
of the cursor (and not necessarily at the end of the line).
Removing it should still cover the same lines of code anyway.
